web/spec/supabase.yml

```diff
@@ -394,14 +394,20 @@ pages:
       ### Update types automatically with GitHub Actions
 
       One way to keep your type definitions in sync with your database is to set up a GitHub action that runs on a schedule.
+      Supabase offers a Swagger OpenAPI specification for the REST data endpoint which you can retrieve definitions for.
 
-      Add a script to your package.json to generate the types:
+      The following script can be run in your terminal to test this.
+      ```
+      npx openapi-typescript https://your-project.supabase.co/rest/v1/?apikey=your-anon-key --output types/database/index.ts
+      ```
+
+      For the GitHub action, add a script to your package.json to generate the types:
 
       ```
-      "update-types": "npx openapi-typescript https://your-project.supabase.co/rest/v1/?apikey=your-anon-key --output types/database/index.ts"
+      "update-types": "npx openapi-typescript \"${SUPABASE_URL}/rest/v1/?apikey=${SUPABASE_ANON_KEY}\" --version=2 --output types/database/index.ts"
       ```
 
-      In your repo, create the file `.github/workflows/update-types.yml`. Add the following snippet into this file to define the action. If there is a change to your definitions, this script will commit the change to your repo.
+      Then in your repo, create the file `.github/workflows/update-types.yml`. Add the following snippet into this file to define the action along with the environment variables. If there is a change to your definitions, this script will commit the change to your repo. ([Reference Article](https://blog.esteetey.dev/how-to-create-and-test-a-github-action-that-generates-types-from-supabase-database#heading-how-to-create-the-github-workflow))
```

Comment:
Was there something missing in the docs that requires a link to the external Reference Article?

Reply:
In that article, there's also a portion on how to test the GitHub action locally with act, thought it might be helpful for users who want to read more in-depth than just using the action.

Reply:
I think it might be useful to add anything that's missing directly here in the docs rather than sending them off to another website

If you want to link to your blog, feel free to add it to the "examples and resources" section of the docs

Reply:
that's true! where's the examples & resources section tho

Reply:
Yah, it's not great having all of this in the yaml file. How about creating a new guide here: https://github.com/supabase/supabase/tree/master/web/docs/guides/database and linking from https://supabase.com/docs/reference/javascript/generating-types to the new guide. Wdyt?

Reply:
I just pushed a commit for your suggestion @thorwebdev , hope it matches what you meant!

Reply:
Aha, that's a great idea to make it a standalone page. It's very close now, but we just need to add it to the sidebar (see `sidebars.js`)

Also, I think this one might be better under the "API" section?

Reply:
@kiwicopple pushed a commit as what you mentioned!

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/35736525/163529422-ec38027e-53ab-4792-8128-69324877adab.png">